### ISSUE TITLE
Touch Controls on Linux

### ISSUE THREAD

**Issue by Henry-Hiles** [divVerent/aaaaxy#414]
Hi! I love AAAXY, and installed it on my linux phone, both from flathub and natively, but I was sad to find that there aren't any touch controls. If they could be added, it would be great! Thanks!

**Comment by Henry-Hiles** [comment]
I see, thanks. Someone *did* make a fork of glfw that has touch support, but afaik it has not been merged. I've seen other apps that use glfw transitioning to sdl3 instead, but as a multiplatform toolkit in and of itself I assume that wouldn't work for ebitengine. Would you like me to close this for now, as it doesn't seem possible?

**Comment by Henry-Hiles** [comment]
Nevermind, looks like that PR is for windows only anyways :/

**Comment by divVerent** [comment]
Either way, it requires added GLFW APIs, so once the Windows one is done, the others will be easy to add. And in the meantime running the windows EXE in WINE is definitely more performant and convenient than the Android or HTML5 approaches.

Once GLFW adds touch APIs, by the PR you linked or another, I can also help implementing the Ebitengine and GLFW sides of the same to also work on Linux (info for that is on https://www.x.org/wiki/Development/Documentation/Multitouch/). I lack the expertise to do it for macOS,

**Comment by Henry-Hiles** [comment]
> Either way, it requires added GLFW APIs, so once the Windows one is done, the others will be easy to add. And in the meantime running the windows EXE in WINE is definitely more performant and convenient than the Android or HTML5 approaches.
> 
> Once GLFW adds touch APIs, by the PR you linked or another, I can also help implementing the Ebitengine and GLFW sides of the same to also work on Linux (info for that is on https://www.x.org/wiki/Development/Documentation/Multitouch/). I lack the expertise to do it for macOS,

That x11 stuff won't work unless AAAXY runs through xwayland though, right?

**Comment by divVerent** [comment]
That I don't actually know, as I use Wayland on all my computers by now. However, even if that is the case, that's still a step forward.

But that's also another issue - Ebitengine has no native Wayland support yet. Maybe gonna happen some day - but right now rest assured everything's well tested through Xwayland.
